include/boost/json/system_error.hpp

```diff
@@ -28,6 +29,9 @@ using error_condition = boost::system::error_condition;
 /// The type of system error thrown by the library.
 using system_error = boost::system::system_error;
 
+template< class T >
+using result = boost::system::result<T>;
```

Comment:
Boost.JSON should follow the model used in Boost.URL. You can probably just copy this file:
https://github.com/CPPAlliance/url/blob/develop/include/boost/url/error_types.hpp

and change the existing JSON headers to include this file instead. Keep the original JSON headers if they overlap and just edit them to only include this new file.